compiler/src/dotty/tools/dotc/transform/MixinOps.scala

```diff
@@ -24,7 +24,7 @@ class MixinOps(cls: ClassSymbol, thisPhase: DenotTransformer)(implicit ctx: Cont
       name = member.name.stripScala2LocalSuffix,
       flags = member.flags &~ Deferred,
       info = cls.thisType.memberInfo(member)).enteredAfter(thisPhase).asTerm
-    res.addAnnotations(member.annotations)
+    res.addAnnotations(member.annotations.filter(_.symbol != defn.TailrecAnnot))
```

Comment:
Sorry, I missed this. Is this needed if you move `TailRec` after `Mixin` within the same group?

Reply:
Even more so. But right now it's necessary above all for *extension methods* (for `AnyVal`s, which also use this `forwarder()` method). In general, when creating a forwarder of any sort, we must remove the `@tailrec` annotation.

Previously, no forwarder was ever created before `TailRec` ran. But now that `TailRec` runs after the creation of extension methods, we must ensure that `forwarder` drops `@tailrec`).